tests/integration/test_analysisd/test_rootcheck/conftest.py

```diff
@@ -0,0 +1,65 @@
+# Copyright (C) 2015-2023, Wazuh Inc.
+# Created by Wazuh, Inc. <info@wazuh.com>.
+# This program is free software; you can redistribute it and/or modify it under the terms of GPLv2
+
+import pytest
+import time
+
+from wazuh_testing.constants.paths.logs import WAZUH_LOG_PATH
+from wazuh_testing.constants.paths.sockets import QUEUE_DB_PATH
+from wazuh_testing.modules.modulesd import patterns
+from wazuh_testing.tools.monitors.file_monitor import FileMonitor
+from wazuh_testing.utils import callbacks
+from wazuh_testing.utils.manage_agents import remove_agents
+from wazuh_testing.utils.services import control_service
+from wazuh_testing.tools.db_administrator import DatabaseAdministrator
+from wazuh_testing.tools.simulators.agent_simulator import connect, create_agents
+
+def inject_rootcheck_events(agents):
+    injectors = []
+
+    for agent in agents:
+        agent.modules['rootcheck']['status'] = 'enabled'
+        _, injector = connect(agent)
+        injectors.append(injector)
+        agents.append(agent)
+
+    # Let rootcheck events to be sent for 60 seconds
+    time.sleep(60)
+
+    for injector in injectors:
+        injector.stop_receive()
+
+    # Service needs to be stopped otherwise db lock will be held by Wazuh db
+    control_service('stop')
+
+def retrieve_rootcheck_rows(agent_id):
```

Comment:
Functions that use the database should be part of the framework. Check if you can use this function (https://github.com/wazuh/qa-integration-framework/pull/14/files#diff-39a6035a005a36ca7752e4c09be145e135a417328ec35fb237b9ea03a5581f44R98) or add a new generic one.